### PR TITLE
Added feature to change server URL in-app

### LINKED ISSUE #73: Create field in Settings menu for base URL
Allows for easily switching between our Staging server and official server for easy testing without recompiling the app.

### PR DESCRIPTION
**Pull request by atharvasathe**
tl;dr: Added a text box in app settings to set the server URL and reset it to default. closes #73 

Previously, in 'strings.xml' there was a hard coded URL for every single server URL (announcements, buses, etc.); all of which had to be changed in order to test changes in staging. Now there is a base URL and strings to concatenate (shuttletracker.app + /buses) for ever single user. Added a text box in app to allow users to modify the base URL and still allow for proper functionality throughout the app with the string concatenation. 

Used Android 'SharedPreferences' interface to store the base URL and retrieve it anywhere throughout the app. [SharedPreferences ](https://developer.android.com/reference/android/content/SharedPreferences)can store key-value pairs that can be read from anywhere in the code and allows a 'default value' in case the preference has not been changed yet. In settings, a reset button has been added to reset the URL to the default production URL.

### CODE REVIEW

**Comment by Gerzer** [comment]
@anugrahaawasthi Could you take a look at this pull request? It has been sitting for nearly a month now without being reviewed, let alone merged.

**Review by anugrahaawasthi — APPROVED** [review]
Looks great!
